src/python/pants/engine/process.py

```diff
@@ -183,19 +187,32 @@ class FallibleProcessResult:
     stderr_digest: FileDigest
     exit_code: int
     output_digest: Digest
+    platform: Platform
+    metadata: ProcessResultMetadata = field(compare=False, hash=False)
 
 
 @dataclass(frozen=True)
-class FallibleProcessResultWithPlatform:
-    """Result of executing a process which might fail, along with the platform it ran on."""
-
-    stdout: bytes
-    stdout_digest: FileDigest
-    stderr: bytes
-    stderr_digest: FileDigest
-    exit_code: int
-    output_digest: Digest
-    platform: Platform
+class ProcessResultMetadata:
+    """Metadata for a ProcessResult, which is not included in its definition of equality."""
+
+    # The execution time of the process, in milliseconds, or None if it could not be captured
+    # (since remote execution does not guarantee its availability).
+    total_elapsed_ms: int | None
```

Comment:
Maybe use `datetime.timedelta` instead to represent duration?

Reply:
Ahm, yea... that would make it more challenging to construct from the Rust side. Will pass for now.